app/src/main/java/com/door43/translationstudio/newui/publish/PublishFragment.java

```diff
@@ -347,6 +373,101 @@ public void onClick(View v) {
         }
     }
 
+    /**
+     * display the upload details
+     * @param destinationMessage
+     */
+    private void showDetails(String destinationMessage) {
+        AlertDialog dialog = new AlertDialog.Builder(getActivity())
+                .setTitle(R.string.project_uploaded)
+                .setMessage(destinationMessage)
+                .setPositiveButton(R.string.dismiss, null)
+                .show();
+        TextView textView = (TextView) dialog.findViewById(android.R.id.message);
+        textView.setMaxLines(5);
+        textView.setScroller(new Scroller(getActivity()));
+        textView.setVerticalScrollBarEnabled(true);
+        textView.setMovementMethod(new ScrollingMovementMethod());
+    }
+
+    /**
+     * put clickable message in dialog
+     * @param clickableMessage
+     * @param dlg
+     */
+    private void applyClickableMessageToDialog(final SpannableString clickableMessage, final CustomAlertDialog dlg) {
+        Handler hand = new Handler(Looper.getMainLooper());
+        hand.post(new Runnable() { // wait until dialog has been created
+            @Override
+            public void run() {
+                try {
+                    Dialog dialog = dlg.getDialog();
+                    TextView textView = (TextView) dialog.findViewById(R.id.dialog_content);
+                    textView.setText(clickableMessage);
+                    textView.setMovementMethod(LinkMovementMethod.getInstance());
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    /**
+     * make selected text clickable
+     * @param textToClick
+     * @param entireText
+     * @param clickableSpan
+     * @return
+     */
+    private SpannableString getClickableText(String textToClick, String entireText, ClickableSpan clickableSpan) {
+        int startIndex = entireText.indexOf(textToClick);
+        int lastIndex;
+        if(startIndex < 0) { // if not found
+            startIndex = 0;
+            lastIndex = textToClick.length();
+        } else {
+            lastIndex = startIndex + textToClick.length();
+        }
+
+        SpannableString clickable = new SpannableString(entireText);
+        clickable.setSpan(clickableSpan,startIndex,lastIndex, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE); // set click
+        clickable.setSpan(new UnderlineSpan(),startIndex,lastIndex,0); // underline
+        clickable.setSpan(new StyleSpan(Typeface.BOLD),startIndex,lastIndex,0); // make bold
+        return clickable;
+    }
+
+    private void openBrowserToUrl(String publishedUrl) {
```

Comment:
we really don't need to place this in a method since it's only used once and it's only two lines. It's just making the code more complicated.